xbmc/cores/VideoPlayer/DVDClock.cpp

```diff
@@ -242,7 +242,7 @@ int CDVDClock::UpdateFramerate(double fps, double* interval /*= NULL*/)
 
   CSingleLock lock(m_speedsection);
 
-  double weight = MathUtils::round_int(rate) / (double)MathUtils::round_int(fps);
+  double weight = rate / fps;
```

Comment:
there are more MathUtils::round_int here. all can be removed

Reply:
The 3 remaining rounds are conceptional

Reply:
they can be removed by replacing the overcomplicated logic by something very simple. the rule is that we want to test if resulting change of speed is within the limits if max m_maxspeedadjust.

speedadjust = (refreshrate mod fps) / refreshrate

example: with 24 fps video on 50 hz monitor

speedadjust = (50 mod 24) / 50 = 0.04

Reply:
you can use `std::fmod` which is the modulus function for double